### PR TITLE
feat(lockfile)!: Smaller lockfile

### PR DESCRIPTION
**Pull request by Natoandro**
### Describe your change
- Use string type for platform instead of a tuple
- Deduplicate task dependencies
- Use relative file urls

### Motivation and context

[MET-393](https://linear.app/metatypedev/issue/MET-393/stabilize-lockfile)

### Migration notes

- If ghjk fails, remove everything from `.ghjk/` directory before re-running the command.

### Checklist

- [ ] The change come with new or modified tests
- [ ] Hard-to-understand functions have explanatory comments
- [ ] End-user documentation is updated to reflect the change

### CODE REVIEW

File: mod.ts [hunk 1]
```diff
@@ -64,33 +63,68 @@ export function install(...configs: InstallConfigFat[]) {
...
+      if (!deep_eql(registered, install)) {
+        throw new Error(
+          `task install already registered with different config: ${install.port.name}`,
+        );
+      }
+    } else {
+      taskDeps[key] = install;
+    }
+    res.push(key);
+  }
```
Comment (Yohe-Am) [inline mod.ts:81]: Would this section not prevent different tasks from relying on different versions of the same port?

Comment (Natoandro) [reply mod.ts]: Fixed!

File: host/types.ts [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
 
 const serializedConfig = zod.object(
   {
     modules: zod.array(moduleValidators.moduleManifest),
+    globalEnv,
   },
 );
 
 export type SerializedConfig = zod.infer<typeof serializedConfig>;
+export type GlobalEnv = zod.infer<typeof globalEnv>;
```
Comment (Yohe-Am) [inline host/types.ts:18]: This type, an unfortunate compromise. As more modules get implemented, any config sections they'll want to share will probably need to put here.

File: modules/ports/sync.ts [hunk 3]
```diff
@@ -429,7 +431,11 @@ export async function buildInstallGraph(
   const foundInstalls: GraphInstConf[] = [];
 
   // collect the user specified insts first
-  for (const inst of portsConfig.installs) {
+  for (const instHash of portsConfig.installs) {
+    const inst = env.installs[instHash];
```
Comment (Yohe-Am) [inline modules/ports/sync.ts:435]: Maybe move this section to `processManifest`? That way, we can keep the `GlobalEnv` an implementation detail of the serialization/lockfile?

File: modules/tasks/types.ts [hunk 4]
```diff
@@ -1,29 +1,43 @@
...
 const taskName = zod.string().regex(/[^\s]/);
 
-const taskEnv = zod.object({
-  installs: portsValidators.installConfigFat.array(),
-  allowedPortDeps: zod.record(
-    zod.string(),
-    portsValidators.allowedPortDep,
-  ),
+const taskEnvBase = zod.object({
+  installs: portName.array(),
```
Comment (Yohe-Am) [inline modules/tasks/types.ts:10]: This is a list of hashes now, right?

**Review by Yohe-Am — APPROVED** [review]
LGTM! Good to have someone helping with jkjk

**Review by Yohe-Am — APPROVED** [review]

**Pull request merged**